src/verso-manual/VersoManual/Markdown.lean

```diff
@@ -241,23 +258,37 @@ where
 open Verso.Doc.Elab
 
 /--
-Updates the active sections given a new header with `level`.
+Close all sections that have a markdown header level that is greater
+than or equal to {name}`level`, to prepare the state for pushing new a
+part at level {name}`level`.
+
+We close a frame in the `.partContext` of `PartElabM.State` exactly in lockstep
+with dropping the head of `inHeaders` in `MDState`.
 -/
 private partial def closeSections {m} [Monad m]
-    [MonadStateOf PartElabM.State m]
+    [MonadError m] [MonadStateOf PartElabM.State m]
     (level : Nat) : MDT m b i Unit := do
   let hdrs := (← getThe MDState).inHeaders
   match hdrs with
-  | [] => modifyThe MDState ({· with inHeaders := [(level, 0)]})
-  | (docLevel, nesting) :: more =>
-    if level ≤ docLevel then
-      if let some ctxt' := (← getThe PartElabM.State).partContext.close default then -- Markdown parser provides no source position
-        modifyThe PartElabM.State fun st => {st with partContext := ctxt'}
-        closeSections level
-      if level < docLevel then
-        modifyThe MDState ({· with inHeaders := more})
-    else
-      modifyThe MDState ({· with inHeaders := (level, nesting + 1) :: hdrs})
+  | [] => pure ()
+  | docLevel :: more =>
+    if docLevel ≥ level then
+      -- `default` here because markdown parser provides no source position
+      let some ctxt' := (← getThe PartElabM.State).partContext.close default
+        |  throwError m!"Failed to close verso part corresponding to markdown section: no parts left"
+      modifyThe PartElabM.State fun st => {st with partContext := ctxt'}
+      modifyThe MDState ({· with inHeaders := more})
+      closeSections level
+
+/--
+In our header mapping bookkeeping, create a new section with a new markdown header with level {name}`level`.
```

Comment:
```suggestion
In our header mapping bookkeeping, creates a new section with a new Markdown header with level {name}`level`.
```
Lean docstrings should use the indicative mood, rather than the imperative. (Other languages are split on this point, we decided to go with the Rust convention)

Reply:
Great, good to know, I was operating under the influence https://leanprover-community.github.io/contribute/commit.html which indeed specifies the opposite convention for commit messages. Nothing wrong with different conventions applying in different contexts.